File: security-platform/agents/ios/Sources/SecurityPlatform/AgentConfig.swift
import Foundation

/// Configuration for the Security Platform Agent.
public struct AgentConfig {
    public var controlPlaneURL: String
    public var authKey: String?
    public var serviceName: String
    public var environment: String
    public var namespace: String
    public var otlpEndpoint: String?
    public var telemetry: TelemetryConfig
    public var redactionRules: [RedactionRule]
    public var policy: PolicyConfig
    public var security: SecurityConfig

    public init(
        controlPlaneURL: String = "https://api.securityplatform.com",
        authKey: String? = nil,
        serviceName: String = "ios-service",
        environment: String = "production",
        namespace: String = "default",
        otlpEndpoint: String? = nil,
        telemetry: TelemetryConfig = TelemetryConfig(),
        redactionRules: [RedactionRule] = [],
        policy: PolicyConfig = PolicyConfig(),
        security: SecurityConfig = SecurityConfig()
    ) {
        self.controlPlaneURL = controlPlaneURL
        self.authKey = authKey
        self.serviceName = serviceName
        self.environment = environment
        self.namespace = namespace
        self.otlpEndpoint = otlpEndpoint
        self.telemetry = telemetry
        self.redactionRules = redactionRules
        self.policy = policy
        self.security = security
    }

    public struct TelemetryConfig {
        public var batchSize: Int
        public var batchTimeout: String
        public var exportTimeout: String
        public var maxQueueSize: Int

        public init(
            batchSize: Int = 100,
            batchTimeout: String = "5s",
            exportTimeout: String = "30s",
            maxQueueSize: Int = 2048
        ) {
            self.batchSize = batchSize
            self.batchTimeout = batchTimeout
            self.exportTimeout = exportTimeout
            self.maxQueueSize = maxQueueSize
        }
    }

    public struct PolicyConfig {
        public var mode: String // "observe" or "block"
        public var autoEnableBlocking: Bool
        public var observePeriodHours: Int
        public var rules: [PolicyRule]

        public init(
            mode: String = "observe",
            autoEnableBlocking: Bool = false,
            observePeriodHours: Int = 48,
            rules: [PolicyRule] = []
        ) {
            self.mode = mode
            self.autoEnableBlocking = autoEnableBlocking
            self.observePeriodHours = observePeriodHours
            self.rules = rules
        }
    }

    public struct SecurityConfig {
        public var mtlsEnabled: Bool
        public var certificatePath: String?
        public var keyPath: String?
        public var caBundlePath: String?

        public init(
            mtlsEnabled: Bool = false,
            certificatePath: String? = nil,
            keyPath: String? = nil,
            caBundlePath: String? = nil
        ) {
            self.mtlsEnabled = mtlsEnabled
            self.certificatePath = certificatePath
            self.keyPath = keyPath
            self.caBundlePath = caBundlePath
        }
    }
}

File: security-platform/agents/dotnet/SecurityPlatform.Agent/AgentConfig.cs
using System.Collections.Generic;

namespace SecurityPlatform.Agent
{
    /// <summary>
    /// Configuration for the Security Platform Agent.
    /// </summary>
    public class AgentConfig
    {
        public string ControlPlaneUrl { get; set; } = "https://api.securityplatform.com";
        public string? AuthKey { get; set; }
        public string? ServiceName { get; set; } = "dotnet-service";
        public string? Environment { get; set; } = "production";
        public string? Namespace { get; set; } = "default";
        public string? OtlpEndpoint { get; set; }
        public TelemetryConfig? Telemetry { get; set; } = new TelemetryConfig();
        public List<RedactionRule>? RedactionRules { get; set; } = new List<RedactionRule>();
        public PolicyConfig? Policy { get; set; } = new PolicyConfig();
        public SecurityConfig? Security { get; set; } = new SecurityConfig();

        public class TelemetryConfig
        {
            public int BatchSize { get; set; } = 100;
            public string BatchTimeout { get; set; } = "5s";
            public string ExportTimeout { get; set; } = "30s";
            public int MaxQueueSize { get; set; } = 2048;
        }

        public class PolicyConfig
        {
            public string Mode { get; set; } = "observe"; // "observe" or "block"
            public bool AutoEnableBlocking { get; set; } = false;
            public int ObservePeriodHours { get; set; } = 48;
            public List<PolicyRule> Rules { get; set; } = new List<PolicyRule>();
        }

        public class SecurityConfig
        {
            public bool MtlsEnabled { get; set; } = false;
            public string? CertificatePath { get; set; }
            public string? KeyPath { get; set; }
            public string? CaBundlePath { get; set; }
        }
    }
}
